packages/modules/viewer-react/src/components/iModel/IModelLoader.tsx

```diff
@@ -71,16 +58,17 @@ const Loader: React.FC<ModelLoaderProps> = React.memo(
     loadingComponent,
   }: ModelLoaderProps) => {
     const [error, setError] = useState<Error>();
-    const [finalFrontstages, setFinalFrontstages] =
-      useState<ViewerFrontstage[]>();
-    const [finalBackstageItems, setFinalBackstageItems] =
-      useState<ViewerBackstageItem[]>();
-    const [viewState, setViewState] = useState<ViewState>();
-    const [noConnection, setNoConnection] = useState<boolean>(false);
     const [connection, setConnection] = useState<IModelConnection>();
     const isMounted = useIsMounted();
+    const { finalFrontstages, noConnectionRequired } = useFrontstages(
+      frontstages,
+      defaultUiConfig,
+      viewportOptions,
+      viewCreatorOptions,
+      blankConnectionViewState
+    );
 
-    useUiProviders(uiProviders, defaultUiConfig);
+    useUiProviders(uiProviders, defaultUiConfig, backstageItems);
```

Comment:
do we need the useEffect in L76 for settingError? couldnt we just do that on L165?

Reply:
you can disregard this, i really need to look at the error-handling pkg and clean it up